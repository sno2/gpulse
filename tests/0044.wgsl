fn f() {
    var a: i32 = 20;
    a++;
    // Now a contains 21
    a--;
    // Now a contains 20
}
