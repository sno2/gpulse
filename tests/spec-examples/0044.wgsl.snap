ast.Node{ .fn_decl = ast.FnDecl{ .name = ast.Span{ .start = 3, .end = 4 }, .params = {  }, .ret = null, .scope = { ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 17, .end = 18 }, .access_mode = ast.AccessMode.read_write, .addr_space = null, .typ = ast.Node{ .identifier = ast.Span{ .start = 20, .end = 23 } }, .value = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 26, .end = 28 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } }, ast.Node{ .inc = ast.UnaryOp{ .value = ast.Node{ .identifier = ast.Span{ .start = 34, .end = 35 } } } }, ast.Node{ .dec = ast.UnaryOp{ .value = ast.Node{ .identifier = ast.Span{ .start = 68, .end = 69 } } } } } } }
